src/crawlee/http_clients/_httpx.py

```diff
@@ -44,8 +47,17 @@ def headers(self) -> HttpHeaders:
         return HttpHeaders(dict(self._response.headers))
 
     def read(self) -> bytes:
+        if not self._response.is_closed:
+            raise RuntimeError('Use `read_stream` to read the body of the Response received from the `stream` method')
         return self._response.read()
 
+    async def read_stream(self) -> AsyncIterator[bytes]:
```

Comment:
I'm tempted to ask what's going to happen if somebody tries to iterate over the same stream twice "in parallel". On the other hand, it's hard enough to do that on purpose, so maybe we can ignore that case.